src/game.cpp

```diff
@@ -9803,10 +9805,25 @@ bool game::handle_liquid( item &liquid, item * const source, const int radius,
         if( item_index != INT_MIN && create_activity() ) {
             serialize_liquid_target( u.activity, item_index );
         } else if( u.pour_into( *cont, liquid ) ) {
+            if( cont->needs_processing() ) {
+                // Polymorphism fail, have to introspect into the type to set the target container as active.
```

Comment:
Because item location is a well scoped interface and I don't want to expand it for such a tiny corner case.  That's a very subjective call that easily could have gone the other way.